network/yarn/pom.xml

```diff
@@ -86,6 +88,15 @@
               </excludes>
             </filter>
           </filters>
+          <relocations>
```

Comment:
I'm now thinking all the dependencies should be relocated, with the exception of the jni leveldb stuff —not the thing you can shade; not the thing you can isolate in a classloader hierarchy instead.

What I can do there is add a profile, `leveldb-provided` which excludes leveldb in this situation. That's important as leveldb does creep onto the classpath from elsewhere —and while the version in Hadoop currently matches that in this JAR, there's no guarantee of that continuing to hold.